plugins/outputs/clarify/clarify.go

```diff
@@ -0,0 +1,186 @@
+//go:generate ../../../tools/readme_config_includer/generator
+
+package clarify
+
+import (
+	"context"
+	_ "embed"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/clarify/clarify-go"
+	"github.com/clarify/clarify-go/fields"
+	"github.com/clarify/clarify-go/views"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+type Clarify struct {
+	Username        config.Secret   `toml:"username"`
+	Password        config.Secret   `toml:"password"`
+	CredentialsFile string          `toml:"credentials_file"`
+	Timeout         config.Duration `toml:"timeout"`
+	IDTags          []string        `toml:"id_tags"`
+	ClarifyIDTag    string          `toml:"clarify_id_tag"`
+	Log             telegraf.Logger `toml:"-"`
+
+	client *clarify.Client
+}
+
+var errIDTooLong = errors.New("id too long (>128)")
+
+const defaultTimeout = config.Duration(20 * time.Second)
+const allowedIDRunes = `abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890-_:.#+/`
+
+//go:embed sample.conf
+var sampleConfig string
+
+func (c *Clarify) Init() error {
+	if c.Timeout <= 0 {
+		c.Timeout = defaultTimeout
+	}
+	// Not blocking as it doesn't do any http requests, just sets up the necessarry Oauth2 client.
+	ctx := context.Background()
+	switch {
+	case c.CredentialsFile != "":
+		creds, err := clarify.CredentialsFromFile(c.CredentialsFile)
+		if err != nil {
+			return err
+		}
+		c.client = creds.Client(ctx)
+		return nil
+	case !c.Username.Empty() && !c.Password.Empty():
+		username, err := c.Username.Get()
+		if err != nil {
+			return fmt.Errorf("getting username failed: %w", err)
+		}
+		password, err := c.Password.Get()
+		if err != nil {
+			config.ReleaseSecret(username)
+			return fmt.Errorf("getting password failed: %w", err)
+		}
+		creds := clarify.BasicAuthCredentials(string(username), string(password))
+		c.client = creds.Client(ctx)
+		config.ReleaseSecret(username)
+		config.ReleaseSecret(password)
```

Comment:
Let's release the secrets as-early-as-possible
```suggestion
		config.ReleaseSecret(username)
		config.ReleaseSecret(password)
		c.client = creds.Client(ctx)
```